Lib/tty.py

```diff
@@ -15,22 +24,81 @@
 OSPEED = 5
 CC = 6
 
-def setraw(fd, when=TCSAFLUSH):
-    """Put terminal into a raw mode."""
-    mode = tcgetattr(fd)
-    mode[IFLAG] = mode[IFLAG] & ~(BRKINT | ICRNL | INPCK | ISTRIP | IXON)
-    mode[OFLAG] = mode[OFLAG] & ~(OPOST)
-    mode[CFLAG] = mode[CFLAG] & ~(CSIZE | PARENB)
-    mode[CFLAG] = mode[CFLAG] | CS8
-    mode[LFLAG] = mode[LFLAG] & ~(ECHO | ICANON | IEXTEN | ISIG)
+try:
+    CHECK_WINSZ = TIOCGWINSZ
+    CHECK_WINSZ = TIOCSWINSZ
+    HAVE_WINSZ = True
+
+    def getwinsz(fd):
+        """Gets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        s = pack("HHHH", 0, 0, 0, 0)
+        return ioctl(fd, TIOCGWINSZ, s)
+
+    def setwinsz(fd, winsz):
+        """Sets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        ioctl(fd, TIOCSWINSZ, winsz)
+except NameError:
+    HAVE_WINSZ = False
+    getwinsz = None
+    setwinsz = None
+
+def mode_echo(mode, echo=True):
+    """Set/unset ECHO."""
+    if echo:
+        mode[LFLAG] |= ECHO
+    else:
+        mode[LFLAG] &= ~ECHO
+
+def mode_raw(mode):
```

Comment:
`enable_raw()` maybe if we agree with `enable_echo()`?

Reply:
Yes, that was a temporary name. This is supposed to function like `cfmakeraw(3)`. While `enable_raw()` does sound better than `mode_raw()`, we could also change it to `tty.makeraw()` if you permit me to do so, and if there are no licensing issues.

Reply:
How about `mkraw()` instead of `makeraw()`? Reasoning: `mkecho()` seems better than `makeecho()`.

Reply:
Ok

Reply:
I have changed the names to `cfmakeraw()`, `cfmakeecho()`, ... so that they conform to standard naming convention.